### PR TITLE
Fixed active states for nav bar

### PR DESCRIPTION
**Pull request by AndriiDovhopolov-okta**
## Description:
- Active states for navbar were fixed. After discussing with @alexdahl-okta  there was decided that each parent of active link will have bold text and active item will remain bold to: 
image

- Fixed issue with relative links and exact links (active link style was applied to parent part of a link, even if link was not currently active - this why first item of sidebar was always bold, see "Languages & SDKs" and "Add user auth to your angular app"). Screenshot on before state, current state on previous screenshot: 
image

- Fixed "keys" properties for Vue for render directive - now link names are used for "keys" instead of array indexes (when we have non-flat arrays we can not use indexes - several elements will have same "key" properties, this can cause side effects when components are updated) 

- Fixed issue with double active links in Concepts section. 

Before: 

bugged-gif

After:

fixed-gif

- Fixed issue with implementing same set of styles twice. 
Before: 
image
After:
image

- **Is this PR related to a Monolith release?** No

### Resolves:

* [OKTA-320822](https://oktainc.atlassian.net/browse/OKTA-320822)

### CODE REVIEW

File: packages/@okta/vuepress-theme-prose/components/Sidebar.vue
Comment (bretterer) [inline packages/@okta/vuepress-theme-prose/components/Sidebar.vue:4]: What was the purpose of changing the `:key` item here?

File: packages/@okta/vuepress-theme-prose/components/SidebarItem.vue
Comment (bretterer) [inline packages/@okta/vuepress-theme-prose/components/SidebarItem.vue:13]: What does `exact` do?

File: packages/@okta/vuepress-theme-prose/components/SidebarItem.vue
Comment (bretterer) [inline packages/@okta/vuepress-theme-prose/components/SidebarItem.vue:22]: Same questions

**Review by bretterer — CHANGES_REQUESTED** [review]
Left some comments before approval

File: packages/@okta/vuepress-theme-prose/components/Sidebar.vue
Comment (AndriiDovhopolov-okta) [reply packages/@okta/vuepress-theme-prose/components/Sidebar.vue:4]: @bretterer mentioned that in PR description:
>when we have non-flat arrays using indexes is not a best solution - several elements will have same "key" properties, this can cause side effects when components are updated and re-rendered. 

In detail - Vue optimizes DOM operations using key properties(basically, as any other framework, such as Angular or React). When Vue see several elements with same keys it will not be happy about that, will get confused and will swap whole DOM representation, instead of particular element with particular key ID. 

There is a link to docs, it's not super in-depth, but explains better than me
https://vuejs.org/v2/guide/list.html#Maintaining-State

File: packages/@okta/vuepress-theme-prose/components/SidebarItem.vue
Comment (AndriiDovhopolov-okta) [reply packages/@okta/vuepress-theme-prose/components/SidebarItem.vue:13]: Here is link for Vue documentation  - https://router.vuejs.org/api/#exact

In general, `exact` will apply `router-link-active` class (default class name provided by Vue) ONLY if the whole current URL will be exactly equal to `:to` directive of `<router-link>`

E.g. we have such URL structure: 

- /parent/
- /parent/first-child/
- /parent/second-child/
- .....
- /parent/nth-child/

`<router-link>` provides `router-link-active` class and URL tracking feature from the box by default. If we will not use `exact` directive, `<router-link>` will treat /parent link as active link, even if current active link is /parent/nth-child, because in provided structure all routes from /first-child/ to /nth-child/ will be treated as siblings and /parent/ will be treated as parent.  

This why there was `router-link-active` class added to each first item in each sidebar sections (Concepts, Languages and SDKs, References). 

Also, due to out-of-the box default `<router-link>` features there is no need to manually add or remove classes to/from this component in Vue, see issue with dual class sets in description of PR.

**Comment by alexdahl-okta** [comment]
Looks great to me! 🎉

File: packages/@okta/vuepress-theme-prose/components/SidebarItem.vue
Comment (AndriiDovhopolov-okta) [reply packages/@okta/vuepress-theme-prose/components/SidebarItem.vue:22]: Look up to previous answer.

File: packages/@okta/vuepress-theme-prose/components/SidebarItem.vue
Comment (oleksandrpravosudko-okta) [inline packages/@okta/vuepress-theme-prose/components/SidebarItem.vue:17]: nit: since we have figured `router-link-active` is a reserved class name in `router-link`, let's pick a different name (for example, `link-ancestor-active`)

**Comment by oleksandrpravosudko-okta** [comment]
this might be a good candidate for adding a selenium test (ancestors and selected item are highlighted with bold font)

**Review by bretterer — APPROVED** [review]
LGTM!  Thanks for the explanations!

**Comment by bretterer** [comment]
After merging #1528, I now have a conflict.  can you fix @oleksandrpravosudko-okta?

**Comment by oleksandrpravosudko-okta** [comment]
@bretterer - https://github.com/okta/okta-developer-docs/pull/1531 contains a copy of  this change set (I don't have write access for Andriy's fork)

**Pull request closed without merging**
